test/test_gpu3.cpp

```diff
@@ -9782,33 +9786,49 @@ TEST_F(NVFuserTest, FusionCrossGridInnerReductionSplitGridIteration_CUDA) {
   fusion.addInput(t0);
   fusion.addOutput(t1);
 
-  auto options =
-      at::TensorOptions().dtype(data_type_to_aten(dtype)).device(at::kCUDA, 0);
-  at::Tensor aten_input = at::randn(input_shape, options);
+  // Reference result is calculated by aten in double precision.
+  // Estimated_gmem is 17.18 GBytes, skip if not enough memory.
+  size_t n_elements = reduction_size * iteration_size + iteration_size * 2;
+  size_t estimated_gmem = n_elements * dataTypeSize(dtype);
+  size_t device_free, device_total;
+  cudaMemGetInfo(&device_free, &device_total);
```

Comment:
I think it's safer to use the stats you can get from the caching allocator as that's where we allocate memory.

Reply:
I didn't find an api to get the free GPU memory from CUDACachingAllocator. `cudaMemGetInfo` should be safe as it is also used by `torch.cuda.mem_get_info (https://pytorch.org/docs/stable/generated/torch.cuda.mem_get_info.html)` https://pytorch.org/docs/stable/generated/torch.cuda.mem_get_info.html

Reply:
Does this help?

https://github.com/NVIDIA/Fuser/blob/main/test/utils.h#L384-L410

Reply:
still didn't get it...
I checked https://github.com/pytorch/pytorch/blob/main/c10/cuda/CUDACachingAllocator.h
looks like CachingAllocator doesn't know the total avialbe GPU memory, it just tries to allocate one and returns error if failed.

Reply:
https://github.com/NVIDIA/Fuser/blob/main/test/utils.h#L404